### ISSUE TITLE
Sync all subscriptions?

### ISSUE THREAD

**Issue by danshaw** [meeb/tubesync#245]
If I have a large number of subscriptions to channels, is there anyway to sync all of them rather than me having to add every single subscription as a source?

**Comment by meeb** [comment]
Currently you need to add each one as a source. What would be a more convenient method for you? It wouldn't be hard to add an import tool of some kind.

**Comment by danshaw** [comment]
Yeah, some type of import feature would be great. Maybe a textbox where we could paste in all of the URLs or import them via a text file. Whatever works. 

Having then some type of way to pick a template for the source parameters would also be pretty cool. So I could say that some of them I'd want to keep the videos all the time and some of them I want them to be deleted after a certain period of time.

**Comment by keithcroshaw** [comment]
CSV Import would be nice.

**Comment by jasonmwade** [comment]
One more vote for this. The Google Takeout option is fairly simple and gives you a CSV with your list of channels with the headers (Channel Id,Channel Url,Channel Title). 
If there were some way to take the subscriptions.csv as input and set some global setting defaults, importing that list should be fairly simple from an end-user standpoint.

**Comment by meeb** [comment]
Can you upload a copy of the takeout CSV so I can see an example of it?

**Comment by jasonmwade** [comment]
See attached for a sample.

[subscriptions.csv](https://github.com/meeb/tubesync/files/13376746/subscriptions.csv)

**Comment by meeb** [comment]
Thanks, looks easy enough to import.

**Comment by tcely** [comment]
See also:
- #44 
- #363

**Comment by derekantrican** [comment]
It would be great if TubeSync could implement the YouTube API and call the `/subscriptions` method to "sync" the user's subscriptions periodically (like once an hour or something) https://developers.google.com/youtube/v3/docs/subscriptions/list?apix_params=%7B%22part%22%3A%5B%22contentDetails%22%5D%2C%22mine%22%3Atrue%7D

**Comment by tcely** [comment]
That sounds like a good project for someone else to do.

A small caching web server that TubeSync could query as often as the user wants without tripping the YouTube limits would be very helpful.

@derekantrican I've opened #1095, for your idea. If a web service, as described in that issue, becomes available, then it would be possible to integrate that with TubeSync.

**Comment by derekantrican** [comment]
Ok, after a lot of ChatGPT (and a _TON_ of arguing with the Google oauth approval team about what this service does), I think I set something up: https://ytsubs.app . I wanted to learn about AWS and figured this would be a pretty simple use-case to work with ChatGPT to build this out.

The homepage should be pretty self explanatory, but basically:
1. Users can go to https://ytsubs.app and click "Sign in with Google"
2. Once the auth flow is completed, they'll be presented with an API key and their Google User ID
3. A request can be made to https://ytsubs.app/subscriptions with `?api_key=APIKEY&google_user_id=GOOGLEUSERID` to retrieve the subscriptions (response format is visible on the homepage)

The user's subscriptions are retrieved from YouTube when the https://ytsubs.app/subscriptions endpoint is called _unless_ the last call was less than 12 hours ago - then the user's subscriptions are returned from the cache

Can people on this thread give this a test and let me know if it seems to work well for you?

**Comment by tcely** [comment]
Will do. Thanks! Feedback is in #1095.

**Comment by tcely** [comment]
I've played with `python3 /app/manage.py youtube-add-subscriptions URL` using the web app. It looks like adding sources for new subscriptions will be easy enough.

The new tasks system has a scheduler included, so we could even add the URL to the `local_settings.py` file to create a task that runs that command every hour.
